<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="node_name" default="huace_gnss" />
  <arg name="param_file_name" default="huace" />
  <arg name="output" default="screen" />
  <arg name="respawn" default="false" />
  <arg name="clear_params" default="true" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_imu"
		args="0 0 0 0 0 0 base_link imu" />

	<node pkg="tf2_ros" type="static_transform_publisher" name="tf_gnss"
		args="0 0 0 0 0 0 imu gnss" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_vsm"
		args="0 0 0 0 0 0 imu vsm" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_aux1"
		args="0 0 0 0 0 0 imu aux1" />

  <node pkg="huace_gnss_driver" type="huace_gnss_driver_node" name="$(arg node_name)"
        output="$(arg output)" 
        clear_params="$(arg clear_params)"
        respawn="$(arg respawn)">
    <rosparam command="load" 
              file="$(find huace_gnss_driver)/config/$(arg param_file_name).yaml" />
  </node>
</launch>